js/piemenus.js

```diff
@@ -3450,7 +3450,33 @@ const piemenuBlockContext = (block) => {
     wheel.navItems[2].navigateFunction = () => {
         that.blocks.activeBlock = blockBlock;
         that.blocks.extract();
-        that.blocks.sendStackToTrash(that.blocks.blockList[blockBlock]);
+        if (that.blocks.selectionModeOn){
+            const blocksArray = that.blocks.selectedBlocks;
+            // figure out which of the blocks in selectedBlocks are clamp blocks and nonClamp blocks.
+            const clampBlocks = [];
+            const nonClampBlocks = [];
+
+            for (let i = 0; i < blocksArray.length; i++) {
+                if (that.blocks.selectedBlocks[i].isClampBlock()) {
+                    clampBlocks.push(that.blocks.selectedBlocks[i]);
+                } else if (that.blocks.selectedBlocks[i].isDisconnected()) {
+                    nonClampBlocks.push(that.blocks.selectedBlocks[i]);
+                }
+            }
+            
+            for (let i = 0; i < clampBlocks.length; i++) {
+                that.blocks.sendStackToTrash(clampBlocks[i]);
+            }
+
+            for (let i = 0; i < nonClampBlocks.length; i++) {
```

Comment:
Question: if a non-clamp block is inside a clamp block, does it need to be included in this list? Won't it be deleted when the clamp block is deleted?

Reply:
When we select the blocks, we do so based on the intersection of the drag area and their position on the screen. This way, all the blocks get selected whether they are inside the clamp block or not. Now, while deleting the blocks, we don't have to delete the ones inside the clamp as they will be deleted by the clamp itself. That's why we use isDisconnected() to check if a block is inside a clamp or not. If anything is unclear, please feel free to ask further questions.

Reply:
Right. I missed the "disconnected line. This looks good.

Reply:
Thanks.